examples/vmod_example/src/lib.rs

```diff
@@ -1,60 +1,53 @@
-// import the generated boilerplate
-varnish::boilerplate!();
-
-// even though we won't use it here, we still need to know what the context type is
-use varnish::vcl::ctx::Ctx;
-
-// we now implement both functions from vmod.vcc, but with rust types.
-// Don't forget to make the function public with "pub" in front of them
-
-pub fn is_even(_: &Ctx, n: i64) -> bool {
-    n % 2 == 0
-}
+/// All public functions in a module tagged with `#[varnish::vmod]` will be exported as Varnish VMOD functions.
+/// The name of the module will be the name of the VMOD.
+#[varnish::vmod]
+mod example {
+    /// A simple function that returns true if the number is even, false otherwise.
+    pub fn is_even(n: i64) -> bool {
+        n % 2 == 0
+    }
 
-// in vmod.vcc, n was an optional INT, so here it translates into an Option<i64>
-pub fn captain_obvious(_: &Ctx, opt: Option<i64>) -> String {
-    // we need to first "match" to know if a number was provided, if not,
-    // return a default message, otherwise, build a custom one
-    match opt {
-        // no need to return, we are the last expression of the function!
-        None => String::from("I was called without an argument"),
-        // pattern matching FTW!
-        Some(n) => format!("I was given {n} as argument"),
+    /// This function can be called without arguments, or with an integer:
+    ///
+    /// ```vcl
+    /// set resp.http.Obvious = example.captain_obvious();
+    /// set resp.http.Obvious-Number = example.captain_obvious(42);
+    /// ```
+    ///
+    /// CString is a Rust's UTF-8 encoded null-terminated string,
+    /// making it the most efficient way to pass strings to Varnish.
```

Comment:
that's not exactly true, the best way is to create a `&CStr` on the workspace and return that

Reply:
I wrote it when trying to get CStrings to work as return types, they kinda do, but not worth it now that we use `txt` internally. Removed.